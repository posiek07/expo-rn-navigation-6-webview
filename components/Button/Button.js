import React from 'react';
import styled from 'styled-components';
import {Linking} from 'react-native';
import propTypes from 'prop-types';

const StyledBookButton = styled.View`
  background-color: #e85d04;
  height: 5%;
  width: 80%;
  position: absolute;
  bottom: 6%;
  left: 10%;
  border-radius: 8px;
  display: flex;
  justify-content: center;
  z-index: 10;
`;

const StyledBookButtonText = styled.Text`
  color: white;
  text-align: center;
  font-size: 18px;
  font-weight: bold;
`;

const Button = ({url}) => {
  return (
    <StyledBookButton>
      <StyledBookButtonText
        onPress={() => {
          Linking.openURL(url);
        }}
        accessibilityLabel="Press me">
        Book Ticket
      </StyledBookButtonText>
    </StyledBookButton>
  );
};

Button.propTypes = {
  url: propTypes.string,
};

export default Button;
