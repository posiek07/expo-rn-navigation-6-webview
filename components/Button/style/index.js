import styled from 'styled-components';



export const StyledBookButton = styled.View`
background-color: #e85d04;
height: 5%;
width: 80%;
position: absolute;
bottom: 6%;
left: 10%;
border-radius: 8px;
display: flex;
justify-content: center;
z-index: 10;
`;

export const StyledBookButtonText = styled.Text`
color: white;
text-align: center;
font-size: 18px;
font-weight: bold;
`;