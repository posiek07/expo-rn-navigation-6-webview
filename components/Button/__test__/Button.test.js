import Button from '../Button';
import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components/native';

it('should find the button via Text', async () => {
  const button = renderer.create(<Button />).toJSON();
  expect(button).toMatchSnapshot();
});
