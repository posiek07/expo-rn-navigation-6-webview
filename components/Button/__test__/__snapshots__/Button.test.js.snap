// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should find the button via Text 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e85d04",
        "borderRadius": 8,
        "bottom": "6%",
        "display": "flex",
        "height": "5%",
        "justifyContent": "center",
        "left": "10%",
        "position": "absolute",
        "width": "80%",
        "zIndex": 10,
      },
    ]
  }
>
  <Text
    accessibilityLabel="Press me"
    onPress={[Function]}
    style={
      Array [
        Object {
          "color": "white",
          "fontSize": 18,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
    Book Ticket
  </Text>
</View>
`;
